folium/features.py

```diff
@@ -351,40 +354,63 @@ class GeoJson(Layer):
                 layer.on({
                     mouseout: function(e) {
-                        e.target.setStyle(e.target.feature.properties.style);},
+                        var style = e.target.feature.properties.style
+                        isMarker?e.target.setIcon(new L.{% if this.marker.icon_type %}{{this.marker.icon_type}}{% else %}Icon{% endif %}(style)):e.target.setStyle(style)},
                     mouseover: function(e) {
-                        e.target.setStyle(e.target.feature.properties.highlight);},
+                        var highlight = e.target.feature.properties.highlight
+                        isMarker?e.target.setIcon(new L.{% if this.marker.icon_type %}{{this.marker.icon_type}}{% else %}Icon{% endif %}(highlight)):e.target.setStyle(highlight)},
+                    mouseenter: function(e) {
+                        var highlight = e.target.feature.properties.highlight
+                        isMarker?e.target.setIcon(new L.{% if this.marker.icon_type %}{{this.marker.icon_type}}{% else %}Icon{% endif %}(highlight)):e.target.setStyle(highlight)},
                     click: function(e) {
-                        {{this._parent.get_name()}}.fitBounds(e.target.getBounds());}
+                        var feat = e.target.feature
+                        if(feat.geometry.type!='Point'){
+                        {{this._parent.get_name()}}.fitBounds(e.target.getBounds())
+                        } else {{this._parent.get_name() }}.setView(e.target.getLatLng(),
+                        Object.keys(feat.properties.highlight).includes('zoom')?feat.properties.highlight.zoom:10)
+                        e.target.bringToFront()}
                     });
             };
         {% endif %}
         var {{this.get_name()}} = L.geoJson(
             {% if this.embed %}{{this.style_data()}}{% else %}"{{this.data}}"{% endif %}
-            {% if this.smooth_factor is not none or this.highlight %}
+            {% if this.smooth_factor is not none or this.highlight or this.marker %}
                 , {
                 {% if this.smooth_factor is not none  %}
-                    smoothFactor:{{this.smooth_factor}}
+                    smoothFactor:{{this.smooth_factor}},
                 {% endif %}
-
                 {% if this.highlight %}
-                    {% if this.smooth_factor is not none  %}
-                    ,
-                    {% endif %}
-                    onEachFeature: {{this.get_name()}}_onEachFeature
+                    onEachFeature: {{this.get_name()}}_onEachFeature,
+                {% endif %}
+                {% if this.marker %}
+                    pointToLayer: function (feature, latlng) {
+                        var opts = {{this.marker.options}};
+                        {% if this.marker._name =='marker' and this.marker.icon_type %}
+                        opts.icon=L.{{this.marker.icon_type}}(opts.icon){% endif %}
+                        return L.{{this.marker._name}}(latlng, opts)
```

Comment:
I guess this is why you're changing the `_name` attributes to lowerCamelCase. Maybe it's better to convert here to the wanted format.

Reply:
the problem there isn't a clean string method idiomatic way to handle both styles of `markers`. `AwesomeMarkers` sublcasses their icon to `L.AwesomeMarkers.icon` while the svg vectors map to `L.circleMarker`, `L.circle`. Coupling `self._name` tightly to how the Leaflet constructors seems like the most straightforward way to do it without introducing another variable or conditional template logic.